src/web/src/app/space/detail/component.ts

```diff
@@ -0,0 +1,277 @@
+import { Component, OnInit, ViewChild, Inject, Injectable } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { MatPaginator, PageEvent } from '@angular/material/paginator';
+import { DOCUMENT } from '@angular/common';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+
+import { Space } from '../model';
+import { SpaceInvitation } from '../model';
+import { SpaceLink } from '../model';
+import { SpaceUser } from '../model';
+
+import { SpaceService } from '../service';
+import { UserService } from '../../user/service';
+import { RequestWatcherService } from '../../request-watcher/service';
+
+@Component({
+	selector: 'app-space-detail',
+	templateUrl: './component.html',
+	styleUrls: ['./component.css']
+})
+export class SpaceDetailComponent implements OnInit {
+
+	invitationForm: FormGroup;
+	isInvitationFormHidden: boolean = true;
+
+	linkForm: FormGroup;
+	isLinkFormHidden: boolean = true;
+
+	currentSpace: Space;
+	currentUserId: string = "2";
+	currentSpaceId;
+
+	linksURL: string = "/space/link/";
+
+	invitations: SpaceInvitation[] = [];
+	links: SpaceLink[] = [];
+	users: SpaceUser[] = [];
+
+	invitationsPageSize: number = 7;
+	linksPageSize: number = 7;
+	usersPageSize: number = 10;
+
+	invitationsTotal: number;
+	linksTotal: number;
+	usersTotal: number;
+
+	@ViewChild('invitationsPaginator') invitationsPaginator: MatPaginator;
+	@ViewChild('linksPaginator') linksPaginator: MatPaginator;
+	@ViewChild('usersPaginator') usersPaginator: MatPaginator;
+
+	currentSpaceLoaded = false;
+
+	constructor(
+		private route: ActivatedRoute,
+		@Inject(DOCUMENT) private document: any,
+		private fb: FormBuilder,
+		private space: SpaceService,
+		private user: UserService,
+		public requestWatcher: RequestWatcherService,
+	) {
+		this._initInvitationForm();
+		this._initLinkForm();
+	}
+
+	ngOnInit() {
+		let pageSize = localStorage.getItem('invitationsPaginatorSpaceDetailPageSize');
+		this.invitationsPageSize = pageSize ? parseInt(pageSize) : 7;
+		
+		pageSize = localStorage.getItem('linksPaginatorSpaceDetailPageSize');
+		this.linksPageSize = pageSize ? parseInt(pageSize) : 7;
+
+		pageSize = localStorage.getItem('usersPaginatorSpaceDetailPageSize');
+		this.usersPageSize = pageSize ? parseInt(pageSize) : 10;
+
+		this.currentSpaceId = this.route.snapshot.paramMap.get('id');
+		this.getCurrentSpace(this.currentSpaceId);
+		this.getInvitations(this.invitationsPageSize, 0);
+		this.getLinks(this.linksPageSize, 0);
+		this.getUsers(this.usersPageSize, 0);
+	}
+
+	getCurrentSpace(spaceId: string) {
+		this.space.getById(spaceId)
+			.subscribe(res => {
+				this.currentSpace = res;
+				this.currentSpaceLoaded = true;
+			});
+	}
```

Comment:
Лучше, как и в функциях ниже, перенести this.currentSpaceId внутрь getCurrentSpace
```suggestion
		this.getCurrentSpace();
		this.getInvitations(this.invitationsPageSize, 0);
		this.getLinks(this.linksPageSize, 0);
		this.getUsers(this.usersPageSize, 0);
	}

	getCurrentSpace() {
		this.space.getById(this.currentSpaceId)
			.subscribe(res => {
				this.currentSpace = res;
				this.currentSpaceLoaded = true;
			});
	}
```

Reply:
Да, конечно. Готово.